DFReader.py

```diff
@@ -1122,15 +1089,16 @@ def rewind(self):
 
     def init_arrays(self, progress_callback=None):
         '''initialise arrays for fast recv_match()'''
-        self.offsets = []
-        self.counts = []
+        offsets = []
+        counts = []
+        formats = self.formats
```

Comment:
This would not seem to be related to the PR title?

I mean, probably not a bad change, but...